### ISSUE TITLE
Crash on Sentry.init()

### ISSUE THREAD

**Issue by tuesda** [getsentry/sentry-java#1436]
_Platform:_
- [x]  Android -> If yes, which Device API (and compileSdkVersion/targetSdkVersion/Build tools) version?
- [ ]  Java -> If yes, which Java (and sourceCompatibility/targetCompatibility) version?
- [ ]  Kotlin -> If yes, which Kotlin (and jvmTarget) version?
- [ ]  NDK -> If yes, which NDK/CMake version?
- [ ]  React-Native -> If yes, which version?
- [ ]  Timber -> If yes, which version?
- [ ]  Log4j2 -> If yes, which version?
- [ ]  Logback -> If yes, which version?
- [ ]  Spring -> If yes, which version?

_IDE:_
- [x]  Android Studio -> If yes, which version?
- [ ]  IntelliJ -> If yes, which version?
- [ ]  Other -> If yes, which one?

_Build system:_
- [x]  Gradle -> If yes, which version?
- [ ]  Buck -> If yes, which version?
- [ ]  Bazel -> If yes, which version?
- [ ]  Maven -> If yes, which version?
- [ ]  Other -> If yes, which one?

_Android Gradle Plugin:_
- [x]  Yes -> If yes, which version? `1.7.36`
- [ ]  No

_Sentry Android Gradle Plugin:_
- [x]  Yes -> If yes, which version?
- [ ]  No

_Proguard/R8:_
- [x]  Enabled
- [ ]  Disabled

_Platform installed with:_
- [x] JCenter
- [ ] Bintray
- [ ] Maven Central
- [ ] Manually

The version of the SDK:
**4.3.0**

---
I have the following issue:

The description goes here ...

**Steps to reproduce:**
- launch app

**Actual result:**
- Crash

device info: 
  - OnePlus 7 Pro 
  - sdk version: 30

stack trace

```
time: 1617886036576
msg: java.lang.NullPointerException: Attempt to invoke interface method 'android.os.Bundle android.content.IContentProvider.call(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, android.os.Bundle)' on a null object reference
stacktrace: java.lang.RuntimeException: Unable to create application io.iftech.match.MainApplication: java.lang.NullPointerException: Attempt to invoke interface method 'android.os.Bundle android.content.IContentProvider.call(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, android.os.Bundle)' on a null object reference
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6721)
	at android.app.ActivityThread.access$1300(ActivityThread.java:237)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7661)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:594)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'android.os.Bundle android.content.IContentProvider.call(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, android.os.Bundle)' on a null object reference
	at android.provider.Settings$NameValueCache.getStringForUser(Settings.java:2769)
	at android.provider.Settings$System.getStringForUser(Settings.java:3205)
	at android.provider.Settings$System.getString(Settings.java:3188)
	at android.hardware.SystemSensorManager.parsePackageList(SystemSensorManager.java:161)
	at android.hardware.SystemSensorManager.<init>(SystemSensorManager.java:136)
	at android.app.SystemServiceRegistry$41.createService(SystemServiceRegistry.java:608)
	at android.app.SystemServiceRegistry$41.createService(SystemServiceRegistry.java:604)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java:1705)
	at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java:1385)
	at android.app.ContextImpl.getSystemService(ContextImpl.java:1919)
	at android.content.ContextWrapper.getSystemService(ContextWrapper.java:803)
	at io.sentry.android.core.TempSensorBreadcrumbsIntegration.register(TempSensorBreadcrumbsIntegration.java:8)
	at io.sentry.core.Sentry.init(Sentry.java:24)
	at io.sentry.core.Sentry.init(Sentry.java:5)
	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:6)
	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:3)
	at d.a.c.a.a(InfoCollector.kt:38)
	at io.iftech.match.MainApplication.onCreate(MainApplication.kt:156)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6716)
	... 8 more
```

**Expected result:**
- No Crash

**Comment by marandaneto** [comment]
@tuesda thanks for reporting

can you elaborate more? is it a UI test? can you paste the full stack trace?

**Comment by tuesda** [comment]
1. This is report by sentry and user feedback.
2. it isn't UI test, it is in production env
3. below is a full stacktrack for example

```
java.lang.NullPointerException: Attempt to invoke interface method 'android.os.Bundle android.content.IContentProvider.call(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, android.os.Bundle)' on a null object reference
    at android.provider.Settings$NameValueCache.getStringForUser(Settings.java:2769)
    at android.provider.Settings$System.getStringForUser(Settings.java:3206)
    at android.provider.Settings$System.getString(Settings.java:3189)
    at android.hardware.SystemSensorManager.parsePackageList(SystemSensorManager.java:161)
    at android.hardware.SystemSensorManager.<init>(SystemSensorManager.java:136)
    at android.app.SystemServiceRegistry$41.createService(SystemServiceRegistry.java:610)
    at android.app.SystemServiceRegistry$41.createService(SystemServiceRegistry.java:606)
    at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java:1722)
    at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java:1402)
    at android.app.ContextImpl.getSystemService(ContextImpl.java:1919)
    at android.content.ContextWrapper.getSystemService(ContextWrapper.java:803)
    at io.iftech.android.looker.b.e(Looker.kt:2)
    at io.iftech.android.podcast.app.singleton.c.b.b(LookerInitializer.kt:1)
    at io.iftech.android.podcast.app.singleton.c.b.c(LookerInitializer.kt:2)
    at io.iftech.android.podcast.app.singleton.c.a.b(Initializer.kt:3)
    at io.iftech.android.podcast.app.singleton.PodcastApp.onCreate(PodcastApp.kt:5)
    at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
    at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6715)
    at android.app.ActivityThread.access$1300(ActivityThread.java:237)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:223)
    at android.app.ActivityThread.main(ActivityThread.java:7660)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
java.lang.RuntimeException: Unable to create application io.iftech.android.podcast.app.singleton.PodcastApp: java.lang.NullPointerException: Attempt to invoke interface method 'android.os.Bundle android.content.IContentProvider.call(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, android.os.Bundle)' on a null object reference
    at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6720)
    at android.app.ActivityThread.access$1300(ActivityThread.java:237)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:223)
    at android.app.ActivityThread.main(ActivityThread.java:7660)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by tuesda** [comment]
info may help: Sentry.init() may be invoked in non-main process

**Comment by marandaneto** [comment]
@tuesda which `Context` are you passing during SDK init? how is your SDK init block? which App's lifecycle? I can't really reproduce it so I need some help to figure this out if its actually a bug related to Android 11 or your SDK's code.

**Comment by tuesda** [comment]
I init Sentry in Application.onCreate() method, and the `Context` is Application instance。And I can't reproduce it neither, those crashes are reported on users' devices.

**Comment by marandaneto** [comment]
@tuesda you have said that the SDK can be invoked in non-main process, how `Application.onCreate()` is called in non-main process? that's not possible, I guess.

Also, how is your SDK init block?

I need detailed info here, otherwise, I'll close it as non-reproducible, thanks.

**Comment by tuesda** [comment]
> how Application.onCreate() is called in non-main process? that's not possible, I guess
I set `android:process` attribute of service in AndroidManifest.xml, which means that service will start in second Dalvik VM, and that's why another application instance is created.

> Also, how is your SDK init block?
This is my init block, which is nothing strange

``` kotlin 
SentryAndroid.init(app) { options ->
            options.beforeSend = SentryOptions.BeforeSendCallback { event, _ ->
                if (disabled.not()) event else null
            }
            var dsn: String? = null
            app::class.java.classLoader!!.getResourceAsStream("sentry.properties")
                .use {
                    val props = Properties()
                    props.load(it)
                    dsn = props.getProperty("dsn")
                }
            require(dsn?.isNotEmpty() == true) { "Dsn can't be null!" }
            options.dsn = dsn
            options.isEnableNdk = true
        }
```

I suspect this line crash, which in `TempSensorBreadcrumbsIntegration.register()`. Could you wrap it with try-catch block?

``` java
sensorManager = (SensorManager) context.getSystemService(SENSOR_SERVICE);
```

**Comment by marandaneto** [comment]
@tuesda can you create a sample repro on github exactly on how you do your setup? the main process, 2nd process, a service, how you start the SDK, etc? I can't reproduce nor understand how is your setup yet.

your code snippet is just weird, why reading a `dsn` from `sentry.properties`?
why setting `isEnableNdk` if it's enabled by default?
what is `app` from `SentryAndroid.init(app)`? `app` does not exist in a class that extends `Application`

Please elaborate otherwise I can't figure this out.

**Comment by tuesda** [comment]
This issue isn't related to my init block, the stack trace already shows that crash occurs in `TempSensorBreadcrumbsIntegration.register()`

``` java
sensorManager = (SensorManager) context.getSystemService(SENSOR_SERVICE);
```

Could you wrap `context.getSystemService(SENSOR_SERVICE)` if that doesn't matter?

**Comment by marandaneto** [comment]
@tuesda this is a workaround (that can't even be tested) and not a proper fix, unless you give us enough information as asked on https://github.com/getsentry/sentry-java/issues/1436#issuecomment-827395143 I'm closing this issue as non-reproducible.

tldr: I've tested it with multiple devices, OS versions, different contexts, and services running in a different processes, none of them failed.

happy to reopen and work on it as soon as we get more info about it.

**Comment by tuesda** [comment]
Ok, I appreciate for your response.
